src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py

```diff
@@ -130,6 +161,13 @@ def get_last_signal(self, message_name: str, signal_name: str) -> Optional[Any]:
 
         return None
 
+    def get_collected_messages(self) -> list[CanMessage]:
+        """Get all the messages collected with the context manager
+
+        :return: a list with all the collected messages
+        """
+        return deepcopy(self._messages_collected)
```

Comment:
You probably should clear the `self._messages_collected` list

Reply:
It is clear whenever the context manager is entered.

Reply:
I think also its nicer so get_collected_messages can be called multiple times.